### PR TITLE
DOC: Remove stability page

### LINKED ISSUE #12438: `docs/stability.rst` should be updated
### Description

`docs/stability.rst` hasn't been touched for quite some time now, and I don't see any open pull requests that would update it. Given that an LTS release is upcoming the file should be reviewed.

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to address remove "stability" page and move some fun facts to relevant subpackage landing pages.

Fixes #12438

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: docs/io/fits/index.rst [hunk 1]
```diff
@@ -11,7 +11,8 @@ Introduction
 
 The :mod:`astropy.io.fits` package provides access to FITS files. FITS
 (Flexible Image Transport System) is a portable file standard widely used in
-the astronomy community to store images and tables.
+the astronomy community to store images and tables. This subpackage was
+originally developed as PyFITS.
```
Comment (taldcroft) [inline docs/io/fits/index.rst:15]: For consistency with the rest?
```suggestion
originally developed as ``PyFITS``.
```

**Review by taldcroft — APPROVED** [review]
Looks good! There is a small suggestion but it's entirely optional.

File: docs/io/fits/index.rst (hunk 1 above)
Comment (pllim) [reply docs/io/fits/index.rst:15]: If you want double backticks, it should be `pyfits` because the actual package name is all lowercase.

**Comment by astrofrog** [comment]
We need to backport this to avoid shipping an out of date page with our 5.0 release

File: docs/io/fits/index.rst (hunk 1 above)
Comment (pllim) [reply docs/io/fits/index.rst:15]: @taldcroft , so do you want to keep PyFITS or change to `pyfits` (all lowercase but with backticks)?

Comment (astrofrog) [reply docs/io/fits/index.rst:15]: I think not having backticks is fine here

**Comment by astrofrog** [comment]
I am going to merge this to make sure it is included in the final 5.0 release

**Pull request merged**